### PR TITLE
docs: improve pagination

### LINKED ISSUE #8254: Bug: Pagination `getFirstPageNumber()` & `getLastPageNumber()` returns unexpected result.
### PHP Version

8.2

### CodeIgniter4 Version

4.4.3

### CodeIgniter4 Installation Method

Composer (using `codeigniter4/appstarter`)

### Which operating systems have you tested for this bug?

Windows

### Which server did you use?

apache

### Database

MariaDB 10.4.18

### What happened?

Within the view file for pagination links, the `getFirstPageNumber()` & `getLastPageNumber()` methods don't seem to return the expected results. From the [docs](https://codeigniter4.github.io/CodeIgniter4/libraries/pagination.html#getfirstpagenumber-getlastpagenumber):
> These methods return page numbers to the first and last pages in the result set.

Let's say your `$pager->links()` prints out the following:
```text
1 | 2 | 3 | 4 | 5
```
calling the `getFirstPageNumber()` & `getLastPageNumber()` returns `1` and `5` respectively. 

And when you navigate to the next page, with links as follows:
```text
3 | 4 | 5 | 6 | 7
```
`getFirstPageNumber()` & `getLastPageNumber()` returns `3` and `7` respectively.

### Steps to Reproduce

Follow the docs and create a simple pagination.

### Expected Output

I expected the code to work differently, `getFirstPageNumber()` should return `1` & `getLastPageNumber()` should return the count of the total number of pages, irrespective of the page being viewed.  As seen in the source code: https://github.com/codeigniter4/CodeIgniter4/blob/14b4350b3685d33018ce7ba0c7d0a368f25962ce/system/Pager/PagerRenderer.php#L386-L389
https://github.com/codeigniter4/CodeIgniter4/blob/14b4350b3685d33018ce7ba0c7d0a368f25962ce/system/Pager/PagerRenderer.php#L402-L405
https://github.com/codeigniter4/CodeIgniter4/blob/14b4350b3685d33018ce7ba0c7d0a368f25962ce/system/Pager/PagerRenderer.php#L86-L89

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
Fixes #8254

**Checklist:**
- [x] Securely signed commits
- [] Component(s) with PHPDoc blocks, only if necessary or adds value
- [] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by sammyskills** [comment]
Can it be stated in the docs, how to get the entire dataset's first and last page numbers? Something along the lines of this https://github.com/codeigniter4/CodeIgniter4/issues/8254#issuecomment-1826432990

**Comment by kenjis** [comment]
@sammyskills Added 066bbbd4bdfcfe9416549946327921c8be2865c4

File: user_guide_src/source/libraries/pagination.rst [hunk 1]
```diff
@@ -299,8 +320,12 @@ These methods return the page number for the previous or next pages in relation
 getFirstPageNumber() & getLastPageNumber()
 ------------------------------------------
 
-These methods return page numbers to the first and last pages in the
-result set.
+These methods return the page numbers to the first and last pages in the set of
+links to be displayed.
```
Comment (sammyskills) [inline user_guide_src/source/libraries/pagination.rst:324]: How about this?

```suggestion
These methods return the page numbers of the first and last pages in the set of
links to be displayed. For example, if the set of links to be displayed is something like this::

    3  |  4  |  5  |  6  |  7

``getFirstPageNumber()`` will return 3 while ``getLastPageNumber()`` will return 7.    
```

File: user_guide_src/source/libraries/pagination.rst [hunk 2]
```diff
@@ -299,8 +320,12 @@ These methods return the page number for the previous or next pages in relation
...
+
+.. note:: If you want to get the page numbers to the first and last pages in the
+    result set, the first page number is always 1, and use `getPageCount()`_ to
+    get the last page number.
```
Comment (sammyskills) [inline user_guide_src/source/libraries/pagination.rst:328]: And this:
```suggestion
.. note:: To obtain the page numbers of the first and last pages in the entire
    result set, you can use the following approach: The first page number is always 1, and `getPageCount()`_ can be used to
    retrieve the last page number.
```

**Review by michalsn — APPROVED** [review]

**Pull request merged**